code/modules/mob/living/simple_animal/hostile/ordeal/green/midnight.dm

```diff
@@ -204,9 +243,178 @@
+/obj/effect/temp_visual/helix_minilaser
+	name = "helix of the end mini-laser"
+	icon = 'ModularTegustation/Teguicons/32x64.dmi'
+	icon_state = "pillar_strike"
+	duration = 15
+	color = "#7ac21f"
+
+/obj/effect/temp_visual/helix_minilaser/Initialize()
+	..()
+	addtimer(CALLBACK(src, PROC_REF(Blowup)), 10)
+
+/obj/effect/temp_visual/helix_minilaser/proc/Blowup()
+	playsound(src, 'sound/weapons/fixer/generic/rcorp4.ogg', 15, FALSE, 4)
+	for(var/mob/living/carbon/human/H in src.loc)
+		H.deal_damage(55, BLACK_DAMAGE)
+		to_chat(H, span_userdanger("You're hit by [src.name]!"))
+
+//This laser hits in a 3 tile radius (the epicenter and its adjacent tiles).
+/obj/effect/temp_visual/helix_macrolaser
+	name = "helix of the end macro-laser"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "warning"
+	duration = 21
+	color = "#7ac21f"
+	pixel_x = -32
+	base_pixel_x = -32
+	pixel_y = -32
+	base_pixel_y = -32
+
+/obj/effect/temp_visual/helix_macrolaser/Initialize()
+	..()
+	addtimer(CALLBACK(src, PROC_REF(Blowup)), 20)
+
+/obj/effect/temp_visual/helix_macrolaser/proc/Blowup()
+	playsound(src, 'sound/abnormalities/crying_children/sorrow_shot.ogg', 20, FALSE, 4)
+	for(var/mob/living/carbon/human/H in range(1, src))
+		H.deal_damage(50, BLACK_DAMAGE)
+		to_chat(H, span_userdanger("You're hit by [src.name]!"))
+
+
+//This proc is called to return a list with all the types that should be spawned by the next deployment. DeployBotSquad() trusts whatever list is given to it, so we
+//make sure we don't generate too many mobs in this proc (else midnight could spawn infinite mobs)
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/GenerateBotSquad()
+	var/list/squad = list()
+	var/remaining_spawn_budget = squad_size - active_minions
+	while(remaining_spawn_budget > 0)
+		//We want the last 40% of our budget to be comprised of Green Noons. They are the priority spawn here
+		if(remaining_spawn_budget <= (squad_size * 0.4))
+			squad += /mob/living/simple_animal/hostile/ordeal/green_bot_big/factory
+		else
+		//If we have budget to spare, we can get one of these I guess. Honestly by this point in the shift, they are irrelevant, EXCEPT the Syringe Bots. They are lethal.
+		//These are specifically the /factory subtype because we don't want a lobillion dead bodies to be left behind.
+			squad += pick(list(
+			/mob/living/simple_animal/hostile/ordeal/green_bot/factory,
+			/mob/living/simple_animal/hostile/ordeal/green_bot/syringe/factory,
+			/mob/living/simple_animal/hostile/ordeal/green_bot/fast/factory))
+		remaining_spawn_budget--
+	return squad
+
+//This proc will deploy the remaining spawn budget (squad size - currently active bots) through drop pods.
+/mob/living/simple_animal/hostile/ordeal/green_midnight/proc/DeployBotSquad()
+	//We get the types we want to spawn in a list here
+	var/list/squad = GenerateBotSquad()
+
+	if(squad.len <= 0)
```

Comment:
instead of `list.len` do `length(list)`, its generally better and may or may not be faster. But its accepted as better
```suggestion
	if(length(squad) <= 0)
```